helix-loader/src/lib.rs

```diff
@@ -88,21 +88,25 @@ pub fn log_file() -> PathBuf {
     cache_dir().join("helix.log")
 }
 
-pub fn find_root_impl(root: Option<&str>, root_markers: &[String]) -> Vec<PathBuf> {
-    let current_dir = std::env::current_dir().expect("unable to determine current directory");
-    let mut directories = Vec::new();
-
-    let root = match root {
+/// Get root to search until.
+pub fn search_root(root: Option<&str>) -> PathBuf {
+    let current_dir = || std::env::current_dir().expect("unable to determine current directory");
+    match root {
         Some(root) => {
-            let root = std::path::Path::new(root);
+            let root = Path::new(root);
             if root.is_absolute() {
                 root.to_path_buf()
             } else {
-                current_dir.join(root)
+                current_dir().join(root)
             }
         }
-        None => current_dir,
-    };
+        None => current_dir(),
+    }
+}
+
+pub fn find_root_impl(root: Option<&str>, root_markers: &[String]) -> Vec<PathBuf> {
```

Comment:
I don't quite know why are we not using `Path` here, but I don't feel like changing this here.